pkg/controllers/rke/node-reporter/controller.go

```diff
@@ -157,3 +181,34 @@ func (h *handler) sameCluster(node *v3.Node, machine *capi.Machine) (bool, error
 
 	return rkeCluster.Spec.ManagementClusterName == node.Namespace, nil
 }
+
+func (h *handler) ensureRKENode(handlerFunc func(string, *v3.Node, []*capi.Machine) (*v3.Node, error)) func(string, *v3.Node) (*v3.Node, error) {
+	return func(key string, node *v3.Node) (*v3.Node, error) {
+		if node == nil {
+			return nil, nil
+		}
+
+		uid := node.Status.NodeLabels["rke.cattle.io/machine"]
+		if uid == "" {
+			return node, nil
+		}
+
+		machines, err := h.machineCache.GetByIndex(machineUID, uid)
+		if apierror.IsNotFound(err) {
+			return node, nil
+		} else if err != nil {
+			return node, err
+		}
+
+		return handlerFunc(key, node, machines)
+	}
+}
```

Comment:
This wrapper function approach doesn't seem necessary. Can `ensureRKEnode` just be a function that is used at the top of both handlers?